lib/SILGen/SILGenBridging.cpp

```diff
@@ -61,9 +62,50 @@ emitBridgeNativeToObjectiveC(SILGenFunction &gen,
   auto witnessFnTy = witnessRef->getType();
 
   // Compute the substitutions.
-  ArrayRef<Substitution> substitutions =
-      swiftValueType->gatherAllSubstitutions(
-          gen.SGM.SwiftModule, nullptr);
+  ArrayRef<Substitution> witnessSubstitutions = witness.getSubstitutions();
```

Comment:
This is pretty epic -- I wanted to clean up witness substitutions at some point since I've got a few changes that are blocked on representation issues, and this provides even more motivation.